ophyd/status.py

```diff
@@ -465,9 +470,14 @@ def wait(self, timeout=None):
             indicates that the action itself raised ``TimeoutError``, distinct
             from ``WaitTimeoutError`` above.
         """
+        exception_to_raise = (
```

Comment:
Creating the exception grabs things about the current stack, we should not create it until we need it.